experimental/Schemes/WeilDivisor.jl

```diff
@@ -660,15 +710,124 @@ end
   return coefficient_dict(D)[C] == 1
 end
 
-is_irreducible(D::WeilDivisor) = is_prime(D)
+is_irreducible(D::AbsWeilDivisor) = is_prime(D)
 
 function order_on_divisor(
     f::VarietyFunctionFieldElem,
-    D::WeilDivisor;
+    D::AbsWeilDivisor;
     check::Bool=true
   )
   @check is_prime(D) || error("divisor must be prime")
   I = components(D)[1]
   return order_on_divisor(f, I, check=false)
 end
 
+########################################################################
+# Implement the interface for AbsAlgebraicCycle and WeilDivisor for    
+# ToricDivisor
+#
+# Because of the inclusion order of the files, this has to be postponed
+# until here.
+########################################################################
+
+scheme(td::ToricDivisor) = toric_variety(td)
+
+### Construct the Weil divisor associated to `td`.
+function underlying_divisor(td::ToricDivisor; check::Bool=false)
+  if has_attribute(td, :underlying_divisor)
+    return get_attribute(td, :underlying_divisor)::WeilDivisor
+  end
+  X = scheme(td)
+  generating_divisors = _torusinvariant_weil_divisors(X; check)
+  result = sum(a*D for (a, D) in zip(coefficients(td), generating_divisors))
+  set_attribute!(td, :underlying_divisor=>result)
+  return result
+end
+
+# Compute the ideal sheaves and Weil divisors of the rays of `fan(X)` 
+# according to the "orbit-cone-correspondence", Theorem 3.2.6 in 
+# Cox-Little-Schenk.
```

Comment:
@lkastner : @HereAround and I had a look at your comments in slack today. We did not quite get your point there. Instead I made an attempt to prove that my hack is correct, see below. 

Be aware that the method is **only** used for the torus invariant prime divisors. So things like 3*ray do not occur here. 

Could you confirm whether or not everything is correct now? If you have questions, please reach out to us through the usual channels.